src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -186,11 +198,34 @@ def get_existing_version_info(self, version: str) -> Optional[dict]:
                     generic_sql_error_handler(err=err, role=self.package_role)
                     return None
 
-    def drop_application_before_upgrade(self, policy: PolicyBase, is_interactive: bool):
+    def drop_application_before_upgrade(
+        self, policy: PolicyBase, is_interactive: bool, cascade: bool = False
+    ):
         """
         This method will attempt to drop an application object if a previous upgrade fails.
         """
-        user_prompt = "Do you want the Snowflake CLI to drop the existing application object and recreate it?"
+        if cascade:
+            try:
+                if application_objects := self.get_objects_owned_by_application():
+                    application_objects_str = self._application_objects_to_str(
+                        application_objects
+                    )
+                    cc.message(
+                        f"The following objects are owned by application {self.app_name} and need to dropped:\n{application_objects_str}"
+                    )
+            except ProgrammingError as err:
+                if err.errno != 93079 and ERROR_MESSAGE_093079 not in err.msg:
+                    generic_sql_error_handler(err)
+                cc.warning(
+                    "The application owns other objects but they could not be determined."
+                )
+            what_to_drop = "the existing application and its owned objects"
+        else:
+            what_to_drop = "the existing application object"
+
+        user_prompt = (
+            f"Do you want the Snowflake CLI to drop {what_to_drop} and recreate it?"
```

Comment:
> Do you want the Snowflake CLI to drop the existing application and its owned objects and recreate it?

Nit: maybe clarify that "it" means the application object, so that there will no expectation for the other dropped objects to be necessarily recreated.

Reply:
+1, when the "and its owned objects" form is used, this becomes a bit ambiguous.

Reply:
i felt that since `it` is singular it can only refer to the application and not the objects, but I can change it

Reply:
updated the wording